nrf9160dk-keys/src/main.c

```diff
@@ -0,0 +1,203 @@
+    if ((in_modem | in_prj) == 0)
+    {
+        printk("No credentials found in modem or in Kconfig (ie: prj.conf\n");
+        printk("Nothing to do here!\n\n");
+        return;
+    }
+    if (in_prj)
+    {
+        printk("Press Button 1 to write Golioth credentials from Kconfig (ie: prj.conf) to modem\n");
+    }
+    if (in_modem)
+    {
+        printk("Press Button 2 to ERASE the PSK-ID and PSK credentials with sec_tag=1 from modem (DANGER!)\n");
+    }
+    printk("\n");
+}
+
+int check_credentials(bool *psk_id_in_modem, bool *psk_in_modem, bool *psk_id_in_prj, bool *psk_in_prj)
+{
+    int err;
+    static const char FOUND[] = "FOUND";
+    static const char ABSENT[] = "ABSENT";
+
+    printk("\nChecking credentials...\n");
+    printk("=======================\n");
+
+    /* Check for credentials in modem */
+    err = modem_key_mgmt_exists(1, MODEM_KEY_MGMT_CRED_TYPE_IDENTITY, psk_id_in_modem);
+    if (err) {
+        printk("Error reading PSK-ID from modem: %d\n", err);
+    }
+    else
+    {
+        printk("%10s: PSK-ID with sec_tag=1 stored in modem\n", *psk_id_in_modem ? FOUND : ABSENT);
+    }
+    err = modem_key_mgmt_exists(1, MODEM_KEY_MGMT_CRED_TYPE_PSK, psk_in_modem);
+    if (err) {
+        printk("Error reading PSK from modem: %d\n", err);
+    }
+    else
+    {
+        printk("%10s: PSK with sec_tag=1 stored in modem\n", *psk_in_modem ? FOUND : ABSENT);
+    }
+
+    /* Check for credentials in prj.conf */
+    #ifdef CONFIG_GOLIOTH_SYSTEM_CLIENT_PSK_ID
+    if (strlen(CONFIG_GOLIOTH_SYSTEM_CLIENT_PSK_ID) > 0) *psk_id_in_prj = 1;
+    #endif
+    printk("%10s: Golioth PSK-ID from Kconfig (ie: prj.conf)\n", *psk_id_in_prj ? FOUND : ABSENT);
+
+    #ifdef CONFIG_GOLIOTH_SYSTEM_CLIENT_PSK
+    if (strlen(CONFIG_GOLIOTH_SYSTEM_CLIENT_PSK) > 0) *psk_in_prj = 1;
+    #endif
+    printk("%10s: Golioth PSK from Kconfig (ie: prj.conf)\n", *psk_in_prj ? FOUND : ABSENT);
+
+    return 0;
+}
+
+void main(void)
+{
+    bool psk_id_in_modem = 0;
+    bool psk_in_modem = 0;
+    bool psk_id_in_prj = 0;
+    bool psk_in_prj = 0;
+
+    check_credentials(&psk_id_in_modem, &psk_in_modem, &psk_id_in_prj, &psk_in_prj);
+
+    /* Buttons */
+    button_mask = 0;
+    dk_buttons_init(button_handler);
+
+    show_prompt((psk_id_in_modem | psk_in_modem), (psk_id_in_prj | psk_in_prj));
+
+    while(1)
+    {
+        if (button_mask) {
+            if (button_mask & BUTTON_1)
+            {
+                write_credentials_to_modem(psk_id_in_prj, psk_in_prj);
+            }
+            else if (button_mask & BUTTON_2)
+            {
+                delete_credentials_from_modem(psk_id_in_modem, psk_in_modem);
+            }
+            k_sleep(K_SECONDS(2));  /* Some timeout to signal to user that something important happened */
+            check_credentials(&psk_id_in_modem, &psk_in_modem, &psk_id_in_prj, &psk_in_prj);
+            show_prompt((psk_id_in_modem | psk_in_modem), (psk_id_in_prj | psk_in_prj));
```

Comment:
Nit: Although they're equivalent in this case, it's generally recommended to use `|` only where you intend bitwise OR. Since this is clearly a logical OR, `||` is preferred.

Reply:
Thanks for the tip. That was sloppy and I'm happy to have it pointed out.